src/awkward/_v2/contents/content.py

```diff
@@ -1299,6 +1312,18 @@ def flatten(self, axis=1, depth=0):
         offsets, flattened = self._offsets_and_flattened(axis, depth)
         return flattened
 
+    def to_backend(self, backend):
+        if backend == ak._v2._util.backend[ak.nplike.Numpy]:
+            if isinstance(self.nplike, ak.nplike.Numpy):
+                return self
+            return self._to_backend(ak.nplike.Numpy.instance())
+        elif backend == ak._v2._util.backend[ak.nplike.Cupy]:
+            if isinstance(self.nplike, ak.nplike.Cupy):
+                return self
+            return self._to_backend(ak.nplike.Cupy.instance())
+        else:
+            raise ValueError("Possible values for backends are `cpu` or `cuda`.")
```

Comment:
What if the `Content.to_backend` (public) takes `backend` as a string, then converts it into an `nplike` singleton using the function in _util.py, and then all of the `_to_backend` (private) methods in each Content subclass expects an `nplike` singleton? That way, you'd be able to use the `_to_backend` methods internally when you need to make sure that a `slicer` from

```python
array[slicer]
```

has the same nplike as the `array`. Also, if `_to_backend` is taking an nplike singleton as input, maybe it should be called `_to_nplike`. That defines the layers: "backends" (strings) come from users into the public API, and "nplikes" (singleton objects) are passed around internally.